src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -84,10 +81,7 @@ public static int LastIndexOfAny(ref byte searchSpace, int searchSpaceLength, re
             for (int i = 0; i < valueLength; i++)
             {
                 var tempIndex = LastIndexOf(ref searchSpace, Unsafe.Add(ref value, i), searchSpaceLength);
-                if (tempIndex != -1)
-                {
-                    index = (index == -1 || index < tempIndex) ? tempIndex : index;
-                }
+                index = (index > tempIndex) ? index : tempIndex;
```

Comment:
I think that the JIT will generate sligthly better code if this is written as `if (index > tempIndex) index = tempIndex;`. (You can check the disassembly.)

Reply:
You are right. 

Simple test with SharpLab (I hope this quite near to real JIT, and JIT for x86 is now also RuyJIT):
```c#
public class C {
    public int A(int tempIndex)
    {
     	int index = -1;
        
        index = (index > tempIndex) ? index : tempIndex;
        
        return index;
    }
    
    public int B(int tempIndex)
    {
     	int index = -1;
        
        if (index > tempIndex) index = tempIndex;
        
        return index;
    }
}
```
produces:
```
C.A(Int32)
    L0000: push ebp
    L0001: mov ebp, esp
    L0003: cmp edx, 0xffffffff
    L0006: jl L000c
    L0008: mov eax, edx
    L000a: jmp L000f
    L000c: or eax, 0xffffffff
    L000f: pop ebp
    L0010: ret

C.B(Int32)
    L0000: or eax, 0xffffffff
    L0003: cmp edx, 0xffffffff
    L0006: jge L000a
    L0008: mov eax, edx
    L000a: ret
```